### PR TITLE
Allow rails 6.1

### PR DESCRIPTION
**Pull request by jrafanie**
* Add appraisal so we can loosen the activesupport dependency but test the ones we support
* Bump rubies to support 2.x versions, removing EOL ones
* Allow rails 6.1 or anything less than 7.0

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -0,0 +1,7 @@
+appraise "activesupport-6.0" do
+  gem "activesupport", "6.0"
+end
+
+appraise "activesupport-6.1" do
+  gem "activesupport", "6.1"
+end
```
Comment (kbrock) [inline Appraisals:7]: I like using `~>` and a `for` loop.
```suggestion
%w(6.0.4 6.1.4).each do |as_version|
  appraise "activesupport-#{as_version.split('.').first(2).join}" do
    gem "activesupport", " ~> #{as_version}"
  end
end
```

Stolen from `virtual_attributes`

File: gemfiles/.bundle/config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+BUNDLE_RETRY: "1"
```
Comment (kbrock) [inline gemfiles/.bundle/config:2]: do we want this file?

File: gemfiles/activesupport_6.0.gemfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# This file was generated by Appraisal
```
Comment (kbrock) [inline gemfiles/activesupport_6.0.gemfile:1]: not sure if we want to call this file activesupport, or if we want to just call it `gemfile`

I guess both work but if we change other gems, it may get confusing

File: manageiq-api-client.gemspec [hunk 4]
```diff
@@ -23,12 +23,13 @@ Gem::Specification.new do |spec|
...
   spec.require_paths = ["lib"]
 
+  spec.add_development_dependency "appraisal"
   spec.add_development_dependency "bundler"
   spec.add_development_dependency "manageiq-style"
   spec.add_development_dependency "rake",          ">= 12.3.3"
   spec.add_development_dependency "rspec",         "~> 3.0"
 
-  spec.add_dependency "activesupport", ">= 5.0", "< 6.1"
+  spec.add_dependency "activesupport", ">= 5.0", "< 7.0"
```
Comment (kbrock) [inline manageiq-api-client.gemspec:32]: Can we just drop the max 7.0 statement?

do we want to test for 7.0 since we know we want to migrate there in the not too distant future?

File: gemfiles/activesupport_6.1.gemfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# This file was generated by Appraisal
+
+source "https://rubygems.org"
+
+gem "activesupport", "6.1"
+
+group :test do
+  gem "webmock"
+  gem "simplecov", require: false
+end
```
Comment (kbrock) [inline gemfiles/activesupport_6.1.gemfile:10]: Can we put these into development dependencies and remove from here?

other projects I worked on only had real dependencies here. (e.g.: activesupport, faraday)

**Review by kbrock — COMMENTED** [review]
looking good

File: Appraisals (hunk 1 above)
Comment (jrafanie) [reply Appraisals:7]: yeah, sorry, I actually started with that but felt it was needlessly complex.

File: gemfiles/.bundle/config (hunk 2 above)
Comment (jrafanie) [reply gemfiles/.bundle/config:2]: No, good idea... I think it stuck in there.  I'll drop it.  Good 👁️

File: gemfiles/activesupport_6.0.gemfile (hunk 3 above)
Comment (jrafanie) [reply gemfiles/activesupport_6.0.gemfile:1]: If these were new gems, yes! But, these are old ones, so YAGNI.  We can rename it if we add other differences we want to test with appraisal.

File: manageiq-api-client.gemspec (hunk 4 above)
Comment (jrafanie) [reply manageiq-api-client.gemspec:32]: Well, that's the nice thing with appraisal.  We can easily add new versions by dropping these needlessly pessimistic constraints.  ActiveSupport is so stable in comparison to ActiveRecord that I would be comfortable leaving off the higher version constraint.  We can always add new appraisals for new versions if we want.

File: gemfiles/activesupport_6.1.gemfile (hunk 5 above)
Comment (jrafanie) [reply gemfiles/activesupport_6.1.gemfile:10]: yeah, they were in the Gemfile.  I can put them in the dev dependencies

File: Appraisals (hunk 1 above)
Comment (jrafanie) [reply Appraisals:7]: The `~>` is right though... it's doing 6.1.0 with what I have now.  Thanks.

File: gemfiles/activesupport_6.0.gemfile (hunk 3 above)
Comment (Fryguy) [reply gemfiles/activesupport_6.0.gemfile:1]: I agree with @kbrock here.  The filename is confusing as is.

File: gemfiles/activesupport_6.0.gemfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+source "https://rubygems.org"
+
+gem "activesupport", "6.0"
+
+group :test do
+  gem "webmock"
+  gem "simplecov", require: false
```
Comment (Fryguy) [inline gemfiles/activesupport_6.0.gemfile:9]: Why are these needed?  I would expect the gemspec to hande this for us.

Comment (jrafanie) [reply gemfiles/activesupport_6.0.gemfile:9]: It was in the Gemfile and appraisal copied them here automatically.

https://github.com/ManageIQ/manageiq-api-client/blob/master/Gemfile#L6-L9

I can move these into the gemspec as @kbrock suggests.

File: gemfiles/activesupport_6.0.gemfile (hunk 3 above)
Comment (jrafanie) [reply gemfiles/activesupport_6.0.gemfile:1]: What do you suggest?  ActiveSupport is the only difference for each appraisal.  If you didn't know this depends on rails or activesupport, `gemfile_6.0.gemfile` would be confusing.

It works with virtual attributes because it's called `activerecord-virtual_attributes` so it's obvious.

File: Gemfile [hunk 7]
```diff
@@ -3,11 +3,6 @@ source 'https://rubygems.org'
 # Specify your gem's dependencies in manageiq-api-client.gemspec
 gemspec
 
-group :test do
-  gem "webmock"
-  gem "simplecov", :require => false
-end
```
Comment (jrafanie) [inline Gemfile:9]: This was removed and pushed down into the gemspec as suggested by @kbrock

File: manageiq-api-client.gemspec (hunk 4 above)
Comment (jrafanie) [reply manageiq-api-client.gemspec:34]: I'm open to dropping the upper bound requirement or we can do it later.  It's not a big deal.

File: Appraisals (hunk 1 above)
Comment (jrafanie) [reply Appraisals]: Ok, I updated to use `~>`.  I thought we could do the loop route if we add another appraisal.  It seems more readable to enumerate the two.

File: gemfiles/.bundle/config (hunk 2 above)
Comment (jrafanie) [reply gemfiles/.bundle/config]: done

File: gemfiles/activesupport_6.1.gemfile (hunk 5 above)
Comment (jrafanie) [reply gemfiles/activesupport_6.1.gemfile]: Great idea, done.

File: gemfiles/activesupport_6.0.gemfile (hunk 6 above)
Comment (jrafanie) [reply gemfiles/activesupport_6.0.gemfile]: Done, via @kbrock's suggestion

File: gemfiles/activesupport_6.0.gemfile (hunk 3 above)
Comment (jrafanie) [reply gemfiles/activesupport_6.0.gemfile:1]: Maybe `rails_6.0-gemfile` and `rails_6.1-gemfile` would be appropriate?  I thought AS would be more explicit that it's a much smaller subset of rails that's in play here.

Comment (Fryguy) [reply gemfiles/activesupport_6.0.gemfile:1]: In activerecord-virtual_attributes they do gemfile_6.0.gemspec, but I agree with you that that also doesn't make sense. `rails_6.0.gemfile` seems better.

Comment (Fryguy) [reply gemfiles/activesupport_6.0.gemfile:1]: Actually, I take it back...I had it in my mind that the version of _rails_ was important, but in this case it's literally activesupport and not all of rails.  rails_6.0.gemfile would be confusing.

I'm good with keeping it as is with `activesupport_6.0.gemfile `

Comment (kbrock) [reply gemfiles/activesupport_6.0.gemfile:1]: I like this. then if we had to update json or csv gem that works with the rails version, then it would not feel too out of place

Comment (Fryguy) [reply gemfiles/activesupport_6.0.gemfile:1]: that argument would still make sense with the activesupport naming (i.e. "if we had to update json or csv gem that works with the activesupport version, then it would not feel too out of place")

**Comment by Fryguy** [comment]
I'm gonna merge as is...having the Rails lockdown for < 7 will force us to create an appraisal when we start working on Rails 7 support.  I know the goal is to open it up, but I don't know if we we want it so open that we forget to properly test it.  Then again, I don't do that on more_core 😉

**Comment by Fryguy** [comment]
We can debate and make any other changes in follow up PRs...this will unstick the 6.1 work though for now.
